packages/chain-controller/src/SnapHandlerClient.test.ts

```diff
@@ -0,0 +1,72 @@
+import type { SnapController } from '@metamask/snaps-controllers';
+import type { SnapId } from '@metamask/snaps-sdk';
+import { HandlerType } from '@metamask/snaps-utils';
+
+import { SnapHandlerClient } from './SnapHandlerClient';
+
+describe('SnapHandlerClient', () => {
+  const handleRequest = jest.fn();
```

Comment:
What are your thoughts on recreating this for each test? Otherwise the state of this mock function may bleed into other tests.